src/containers/Tenant/Schema/CreateDirectoryDialog/CreateDirectoryDialog.tsx

```diff
@@ -0,0 +1,116 @@
+    const [create, response] = schemaApi.useCreateDirectoryMutation();
+
+    const resetErrors = () => {
+        setValidationError('');
+        setError('');
+    };
+
+    const handleUpdate = (updated: string) => {
+        setRelativePath(updated);
+        resetErrors();
+    };
+
+    const handleClose = () => {
+        onClose();
+        setRelativePath('');
+        resetErrors();
+    };
+
+    const handleSubmit = async () => {
+        try {
+            const path = `${parentPath}/${relativePath}`;
+            await create({
+                database: parentPath,
+                path,
+            }).unwrap();
+            handleClose();
+            onSuccess(relativePath);
+        } catch (e) {
+            setError(e);
+        }
+    };
+
+    const hasError = Boolean(error);
+    const hasValidationError = Boolean(validationError);
+
+    return (
+        <Dialog open={open} onClose={handleClose} size="s">
+            <Dialog.Header caption={i18n('schema.tree.dialog.header')} />
+            <form
+                onSubmit={(e) => {
+                    e.preventDefault();
+                    const validationError = validateRelativePath(relativePath);
+                    setValidationError(validationError);
+                    if (!validationError) {
+                        handleSubmit();
+                    }
+                }}
+            >
+                <Dialog.Body>
+                    <div className={b('label')}>
+                        <div className={b('description')}>
+                            {i18n('schema.tree.dialog.description')}
+                        </div>
+                        <div>{`${parentPath}/`}</div>
+                    </div>
+                    <TextInput
+                        placeholder={i18n('schema.tree.dialog.placeholder')}
+                        value={relativePath}
+                        onUpdate={handleUpdate}
+                        autoFocus
+                        hasClear
+                        disabled={response.isLoading}
+                        error={hasValidationError}
+                    />
+                    <div className={b('error-wrapper')}>
+                        {hasValidationError && <ResponseError error={validationError} />}
+                    </div>
+                    {hasError && (
+                        <ResponseError
+                            error={error}
+                            defaultMessage={i18n('schema.tree.dialog.invalid')}
+                        />
+                    )}
+                </Dialog.Body>
+                <Dialog.Footer
+                    loading={response.isLoading}
+                    textButtonApply={i18n('schema.tree.dialog.buttonApply')}
+                    textButtonCancel={i18n('schema.tree.dialog.buttonCancel')}
+                    onClickButtonCancel={handleClose}
+                    propsButtonApply={{disabled: hasValidationError, type: 'submit'}}
```

Comment:
```suggestion
                    propsButtonApply={{type: 'submit'}}
```

Reply:
You suggest not to disable Apply button at all? But why?

Reply:
Why do you want to disable it? There is a validation before submission.

Reply:
If it's validation error, we 100% know, that "Apply" won't work.
Or maybe I just like to forbid :)